# Module 03

## Session 02 Supervised Learning Regression

## Simple Linear Regression

### Import data

In [ ]:
import statsmodels.api as sm
import seaborn as sns

### Load data

In [3]:
tips = sns.load_dataset('tips')
tips.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

### Model

In [5]:
# variable
y = tips['tip'] # dependent
X = tips[['total_bill']] # independent

# add intercept to the formula
X = sm.add_constant(X) 

# defining the model
model = sm.OLS(y, X)

# fitting model
result = model.fit()

# print summary
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:                    tip   R-squared:                       0.457
Model:                            OLS   Adj. R-squared:                  0.454
Method:                 Least Squares   F-statistic:                     203.4
Date:                Wed, 07 Jul 2021   Prob (F-statistic):           6.69e-34
Time:                        10:44:59   Log-Likelihood:                -350.54
No. Observations:                 244   AIC:                             705.1
Df Residuals:                     242   BIC:                             712.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.9203      0.160      5.761      0.0

<b>Important output</b>:
1. <b>R-square</b> 
2. <b>Prob (F-statistic)</b> (F-test)
3. <b>coef each variables</b> (β<sub>0</sub> and β<sub>1</sub>) 
4. <b>prob each variables</b> (T-test from β<sub>0</sub> and β<sub>1</sub>)

### Summary interpretation:
Model: <b>y = 0.9203 + 0.1050X</b>
1. <b>R-square</b>: the model can define 45.7% variation of tips
2. <b>Prob (F-statistic)</b>: 6.69e-34 < 0.05, then the total bill significantly affect the tips
3. <b>β<sub>0</sub> and β<sub>1</sub></b>:
> - β<sub>0</sub>: y = β<sub>0</sub> + β<sub>1</sub>X, y when x=0, tips when total bill = 0
> - β<sub>1</sub>: Every 10 dollars total bill increase, then it will increase the tips approximately 1.050 dollars
4. <b>T-test</b>:
> - β<sub>0</sub> p-value: 0.000... < 0.05, then it confirm the model needs intercept
> - β<sub>1</sub> p-value: 0.000... < 0.05, then it confirm the correlation between total bill and tips are significant and the correlations are positive

<b>Note</b>: 
> - <b>F-test</b>: to test wether the independent variable affects the dependent variable
>> - <b>Hypothesis</b>: Independent variable <b>does not</b> affects dependent variable
>> - <b>H<sub>0</sub></b>: β<sub>1</sub> = 0
>> - <b>H<sub>a</sub></b>: β<sub>1</sub> != 0
>> - <b>Rejection criteria</b>: p-value <= α
> - <b>T-test</b>: to test wether β<sub>0</sub> and β<sub>1</sub> are significant and wether β<sub>0</sub> (intercept/constant) are needed
>> - <b>Hypothesis</b>: Model <b>does not</b> need the intercept
>> - <b>H<sub>0</sub></b>: β<sub>0</sub> = 0
>> - <b>H<sub>a</sub></b>: β<sub>0</sub> != 0 or β<sub>0</sub> != 0 or β<sub>0</sub> != 0 
>> - <b>Rejection criteria</b>: 
>>> - <b>One-sided</b>: p-value <= α
>>> - <b>Two-sided</b>: p-value/2 <= α
----------------------